### PR TITLE
[monorepo] fix: upgrade all the python sdk to 3.0.7

### PR DESCRIPTION
**Pull request by Wayonb**
problem: dependabot is failing to upgrade the python sdk to 3.0.7
solution: 1. update Symbol testnet generation hash seed and epoch time 
              2. fix breaking changes where to_datetime moved to NetworkTimestampDatetimeConverter
              3. update imports for classes that moved

### CODE REVIEW

File: tools/vanity/tests/test_AddressGenerator.py [hunk 1]
```diff
@@ -11,7 +11,8 @@
 class AddressGeneratorTest(unittest.TestCase):
 	@staticmethod
 	def _derive_public_key_at(facade, mnemonic, account_index):
-		coin_id = 1 if 'testnet' == facade.network.name else facade.BIP32_COIN_ID
+		bip32_coin_id = 43
```
Comment (Jaguar0625) [inline tools/vanity/tests/test_AddressGenerator.py:14]: use facade.bip32_path ?

File: tools/vanity/tests/test_main.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def _execute(blockchain, patterns):
 
 	def _test_can_generate_addresses(self, blockchain, facade_class):
 		# Act:
+		bip32_coin_id = 43
```
Comment (Jaguar0625) [inline tools/vanity/tests/test_main.py:34]: use facade.bip32_path ?

File: tools/vanity/vanity/AddressGenerator.py [hunk 3]
```diff
@@ -30,7 +30,8 @@ def match_all(self, matcher):
 			thread.join()
 
 	def _match_all_thread(self, matcher):
-		coin_id = 1 if 'testnet' == self.facade.network.name else self.facade.BIP32_COIN_ID
+		bip32_coin_id = 43
```
Comment (Jaguar0625) [inline tools/vanity/vanity/AddressGenerator.py:33]: use facade.bip32_path ?

**Review by Jaguar0625 — CHANGES_REQUESTED** [review]

File: optin/puller/puller/models/NetworkTimeConverter.py [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import datetime
 
-from symbolchain.nem.NetworkTimestamp import NetworkTimestamp as NemNetworkTimestamp
-from symbolchain.symbol.NetworkTimestamp import NetworkTimestamp as SymbolNetworkTimestamp
+from symbolchain.NetworkTimestamp import NetworkTimestampDatetimeConverter
 
-SYMBOL_TESTNET_EPOCH_TIME = datetime.datetime(2021, 11, 25, 14, 0, 47, tzinfo=datetime.timezone.utc)
+SYMBOL_TESTNET_EPOCH_TIME = datetime.datetime(2022, 10, 31, 21, 7, 47, tzinfo=datetime.timezone.utc)
+SYMBOL_MAINNET_EPOCH_TIME = datetime.datetime(2021, 3, 16, 0, 6, 25, tzinfo=datetime.timezone.utc)
+NEM_EPOCH_TIME = datetime.datetime(2015, 3, 29, 0, 6, 25, tzinfo=datetime.timezone.utc)
```
Comment (Jaguar0625) [inline optin/puller/puller/models/NetworkTimeConverter.py:7]: why can't you read these from the Network.[MAINNET|TESTNET] ?

File: tools/vanity/tests/test_AddressGenerator.py (hunk 1 above)
Comment (Jaguar0625) [reply tools/vanity/tests/test_AddressGenerator.py]: i mean isn't this
`[44, coin_id, account_index, 0, 0]`
same as 
`facade.bip32_path(account_index) ?

Comment (Wayonb) [reply tools/vanity/tests/test_AddressGenerator.py]: yup

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**